### DISCUSSION TITLE
Error while compilining ESP32 with bluetooth

### DISCUSSION THREAD

**Discussion by host-felice** [arendst/Tasmota#19663] [Support request] [answered]
I am trying to compiling my own firmware for esp32 adding mitsubishi hvac and bluetooth. I tried with tasmocompiler and with docker-tasmota and from 13.1 I always get an error of NimBLEDevice.h missing.
With 13.0 and dev I was able to.

Any clue?

(I could add the mitsubishi hvac to the esp32-bluetooth, but I know it is not the correct way)

/tmp/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_79_esp32_ble.ino:148:10: fatal error: NimBLEDevice.h: No such file or directory

    Looking for NimBLEDevice.h dependency? Check our library registry!
    CLI > platformio lib search "header:NimBLEDevice.h"
    Web > https://registry.platformio.org/search?q=header:NimBLEDevice.h

#include <NimBLEDevice.h>
^~~~~~~~~~~~~~~~
compilation terminated.
Compiling .pio/build/tasmota32/lib8ec/freetype/smooth/smooth.c.o
*** [.pio/build/tasmota32/src/tasmota.ino.cpp.o] Error 1
========================= [FAILED] Took 48.93 seconds =========================

Environment Status Duration

tasmota32 FAILED 00:00:48.928
==================== 1 failed, 0 succeeded in 00:00:48.928 ====================
Finished. Exit code: 1.

**Comment by barbudor** [comment]
If the compile tries to include NimbleDevice.h then it means that you have USE_BLE_ESP32 defined
Please provide all details how you are compiling
- what variant are you compiling
- what do you have in your user_config_override.h

**Comment by host-felice** [comment]
indeed...

with docker-tasmota

#define USE_MIEL_HVAC
#define USE_MI_ESP32
#define USE_BLE_ESP32

and I am trying to compile tasmota32

With tasmocompiler I just tick the bluetooth option. I asked to the author and he said:
"It looks like there is something wrong when Tasmota v13.1.0 is chosen (v13.0.0 works fine on my side).
Please contact Tasmota Dev Team (via Discord channel) and describe that situation."

I confirm that with 13.0 is fine with me too

**Reply by barbudor** [reply]
Indeed tasmota32 cannot be compiled with USE_BLE_ESP32 without a change in one of the ini to include the Nimble library
Nimble is not enabled by default because it increase the compile time a lot for people who don't need it

I think there is a comment somewhere of a ; to remove
Or you can have a look at how env:tasmota32-bluetooth is defined and report the additional libs in env:tasmota32

**Comment by host-felice** [comment]
got it. I was able to compile the firmware by editing platformio_override.ini and edit this part: 

; *** uncomment the following line if you use Bluetooth or Apple Homekit in a Tasmota32 build. Reduces compile time
;                          lib/libesp32_div

removing the ; in the lib/libesp32_div line

I understand the problem, but now the tasmocompiler should be mod for this. I will send a note to benzino

**Reply by barbudor** [reply]
Great to see you found it
I answered you from my mobile earlier so I wasn't able to look further and point you in the exact location

**Reply by host-felice** [reply]
thanks :-) no problem, I like to study and learn things. I just think the wiki should be changed. At the moment it is written:

When compiling your build add the following to user_config_override.h:

#define USE_BLE_ESP32                // Add support for ESP32 as a BLE-bridge (+9k2? mem, +292k? flash)


It was working till 13.0.0

**Reply by barbudor** [reply]
Yes definitively it should be written and you can propose doc updates by clicking on the pencil button at the top right of the page.

**Comment by host-felice** [comment]
thanks, but I am not really able to. I barely understood what I did to make it possible :-D

**Reply by barbudor** [reply, marked as answer]
Done 

![image](https://github.com/arendst/Tasmota/assets/2996042/0ffbcc41-1fcb-4ca2-b0b1-b4529b2b8d93)
